### PR TITLE
Refactor global providers

### PR DESCRIPTION
**Pull request by agualis**
# Description

We currently use [Roots.ts](https://github.com/balancer-labs/frontend-v2/blob/develop/src/Root.ts) to provide/inject global providers so that they can be used by any component. This solution works but it makes testing setup challenging, as it is really difficult to instantiate a component under test with access those injected providers.

This PR refactors and simplifies the global provider registration by solving the following challenges: 

### vue-query requires setup

`tokens.provider` uses `vue-query`'s `useQuery` that cannot be used outside Vue's setup method so that we cannot use `app.use`  to register this provider. Instead, we are providing  from App.vue setup function (through `globalProvide` helper) so that all the component's hierarchy has access to the global provided content. 

### Dependencies between global providers

`tokens.providers` depends on `userSettings.provider` and `token.lists.provider` but we cannot directly call `useUserSettings` or `useTokenLists` (using inject underneath) because they should be provided from a higher level in the component hierarchy but App.vue is already the root level. Instead, we explicitly pass those dependencies in the `provideTokens` signature.

### Testing the previous setup is challenging

This PR solves the problem by introducing our own version of `vue-composable-tester` which allows more control over the injected providers and plugins. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [x] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

### CODE REVIEW

**Comment by garethfuller** [comment]
Could you resolve the merge conflicts, please?

File: src/plugins/register-plugins.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    .use(vueQuery)
+    .use(Web3Plugin)
+    .use(VueVirtualScroller);
+  return app;
+}
+
+export function registerTestPlugins(app: App) {
+  app.use(blocknative).use(Web3Plugin).use(vueQuery).use(VueVirtualScroller);
+  return app;
+}
```
Comment (garethfuller) [inline src/plugins/register-plugins.ts:25]: Again, my preference would be for this to be in plugins/index.ts so it's clear that it's where the core/init logic is for plugins.

**Review by garethfuller — CHANGES_REQUESTED** [review]

File: src/providers/globalProvide.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { provideUserSettings } from '@/providers/user-settings.provider';
+import { provideTokenLists } from '@/providers/token-lists.provider';
+import { provideTokens } from '@/providers/tokens.provider';
+
+// Provides global providers that can be injected from any component
+export function globalProvide() {
+  const userSettings = provideUserSettings();
+  const tokenLists = provideTokenLists();
+  provideTokens(userSettings, tokenLists);
+}
```
[the remark this answers is not in the document]
Comment (agualis) [reply src/providers/globalProvide.ts:10]: The only drawback is that looking for` index.ts' in your IDE is more difficult but I also agree that the intention of the file is clearer so, change done.

File: src/main.ts [hunk 3]
```diff
@@ -39,15 +33,9 @@ echarts.use([
...
-  .use(router)
-  .use(store)
-  .use(blocknative)
-  .use(vueQuery)
-  .use(Web3Plugin)
-  .use(VueVirtualScroller);
+const app = createApp(App);
 
+registerPlugins(app);
 registerDirectives(app);
```
[the remark this answers is not in the document]
Comment (agualis) [reply src/main.ts:35]: Done.

File: src/plugins/register-plugins.ts (hunk 1 above)
Comment (agualis) [reply src/plugins/register-plugins.ts]: Changed.

**Review by garethfuller — APPROVED** [review]

**Pull request merged**